x-pack/platform/plugins/shared/security/server/saved_objects/access_control_service.ts

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { ISavedObjectTypeRegistry } from '@kbn/core/server';
+import type {
+  AuthorizeObject,
+  CheckAuthorizationResult,
+} from '@kbn/core-saved-objects-server/src/extensions/security';
+import type { AuthenticatedUser } from '@kbn/security-plugin-types-common';
+
+export class AccessControlService {
+  private userForOperation: AuthenticatedUser | null = null;
+
+  constructor() {}
+
+  setUserForOperation(user: AuthenticatedUser | null) {
+    this.userForOperation = user;
+  }
+
+  getTypesRequiringPrivilegeCheck({
```

Comment:
No need to do anything in this PR, just an observation. Something feels a bit off with these two methods: `getTypesRequiringPrivilegeCheck` doesn't trust the consumer that all objects have types that support access controls and checks this on its own using the type registry. However, `enforceAccessControl` trusts consumers that `typesRequiringAccessControl` contains only supported types and doesn't require a type registry. I'm wondering if in the future we either always trust consumers to filter objects before passing them to these methods (and hence don't require a type registry) or perform the check in every method, which might require a type registry directly in the constructor.